src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -789,6 +789,18 @@ void primitive_inst::realloc_if_needed() {
             for (auto& user_inst : get_user_insts()) {
                 reset_user_output_memory(user_inst, dep_memory_ptr(0));
             }
+        } else {
+            // when this inst was not executed at the previous iteration,
+            // Reset output memory becuase current output memory is invalid.
+            if (_no_execution_prev) {
+                if (_outputs[0]) {
+                    for (auto& user_inst : get_user_insts()) {
+                        reset_user_output_memory(user_inst, _outputs[0]);
+                    }
+                }
+                _outputs[0] = nullptr;
+                _max_output_layout_count[0] = 0;
```

Comment:
These two are being used here and there. Could you refactor this as a member function of primitive_inst? (e.g., clear_output_memory)

Reply:
Fixed it. please review it again.